### ISSUE TITLE
Suggestions

### ISSUE THREAD

**Issue by sbonaretti** [drcandacemakedamoore/cleanX#27]
Can you add documentation in the following files?
  - journaline_pipeline.py, starting from line 110  
  - steps.py starting from line 112
  - Many functions in the fils dataframes.py, pydicom_adapter.py, and simpleitk_adapter.py

**Comment by wvxvw** [comment]
Hi Serena.

There's a lot to process, I'll respond to something that's easy to answer:

> Other: In the file steps.py: Why do you have classes and not simply functions?

I believe, you refer to classes such as Acquire, Save etc. There are
multiple reasons for making them into classes:

1. Some of them need to use parameters not supplied by the call site
(i.e. the pipeline). If those were defined as functions, the user
would then have to use something like functools.partial to create
closures with the extra parameters captured. Not impossible, but seems
inconvenient.
2. In Python, functions are global. On the other hand, the pipeline
code relies on the state of the step to figure out certain
information, s.a. what is the cache directory where the images need to
be stored during processing. If all steps of the same kind were
indistinguishable, the pipeline code would have to be a lot more
involved to keep track of what step matches what cache directory.
3. Steps need to be pickled and serialized for command-line use. Since
there isn't a comfortable mechanism to define serialization behavior
for a bunch of functions, especially if something like
functools.partial was used, the pickling behavior would have to be
defined on per-step basis. Again, not impossible, but seems very
inconvenient.
4. Right now, the command-line deserialization of steps relies on all
steps being available from a predefined storage location by simply
knowing their names. The addition to this storage location is
implemented using metaclass. This relieves the user of the library who
wants to create their own step from responsibility to register it to
make it available via command-line, it would be inconvenient to ask
users, in addition to defining a step function to also register it
somewhere.

Bottom line: It might have been possible to make them into functions,
but in doing so, the potential user of the library would have to
understand a lot more about its internals and would have to put a lot
more effort into using it.

It is possible that in the future we'll add functions for creating
steps, thus also removing the need for users to understand how to work
with classes.

**Comment by drcandacemakedamoore** [comment]
Thank you Serena for all of your comments here and elsewhere. I will be adding more documentation to the areas you pointed out starting today. We hope this will allow you to use the package with greater ease.

**Comment by drcandacemakedamoore** [comment]
DOcumentation has been updated, as well as some other issues addressed on the develop branch...soon to be pushed to main.
